nshmdb/nshmdb.py

```
@@ -0,0 +1,279 @@
+"""
+Module to interact with the NSHMDB (National Seismic Hazard Model Database).
+
+This module provides classes and functions to interact with an SQLite database
+containing national seismic hazard model data. It includes functionalities to
+insert fault and rupture data into the database, as well as retrieve fault
+information associated with ruptures.
+
+Classes
+-------
+NSHMDB
+    Class for interacting with the NSHMDB database.
+
+Usage
+-----
+Initialize an instance of NSHMDB with the path to the SQLite database file.
+Use the methods of the NSHMDB class to interact with fault and rupture data
+in the database.
+
+>>> db = NSHMDB('path/to/nshm.db')
+>>> db.get_rupture_faults(0) # Should return two faults in this rupture.
+"""
+
+import dataclasses
+import importlib.resources
+import sqlite3
+from pathlib import Path
+from sqlite3 import Connection
+
+import numpy as np
+
+from nshmdb import fault
+from nshmdb.fault import Fault
+
+
+@dataclasses.dataclass
+class NSHMDB:
+    """Class for interacting with the NSHMDB database.
+
+    Parameters
+    ----------
+        db_filepath : Path
+            Path to the SQLite database file.
+    """
+
+    db_filepath: Path
+
+    def __init__(self, db_filepath: Path):
+        """Create a NSHMDB object
+
+        Parameters
+        ----------
+        db_filepath : Path
+            The path to the nshm database.
+
+        Raises
+        ------
+        ValueError
+            If the database at db_filepath does not exist.
+        """
+        if not db_filepath.exists():
+            raise ValueError(f"DB filepath {str(db_filepath)} does not exist.")
```

Probably should raise a file not found error?